pandas-stubs/core/reshape/tile.pyi

```diff
@@ -1,13 +1,212 @@
+    include_lowest: bool = ...,
+    duplicates: Literal["raise", "drop"] = ...,
+    ordered: bool = ...,
+) -> tuple[Categorical, IntervalIndex]: ...
+@overload
+def cut(
+    x: Index | npt.NDArray | Sequence[int] | Sequence[float],
+    bins: int
+    | Series
+    | Int64Index
+    | Float64Index
+    | Sequence[int]
+    | Sequence[float]
+    | IntervalIndex,
+    right: bool = ...,
+    *,
+    labels: Literal[False],
+    retbins: Literal[False] = ...,
+    precision: int = ...,
+    include_lowest: bool = ...,
+    duplicates: Literal["raise", "drop"] = ...,
+    ordered: bool = ...,
+) -> npt.NDArray[np.intp]: ...
+@overload
+def cut(
+    x: Series,
+    bins: int
+    | Series
+    | Int64Index
+    | Float64Index
+    | Sequence[int]
+    | Sequence[float]
+    | IntervalIndex,
+    right: bool = ...,
+    labels: Literal[False] | Sequence[Label] | None = ...,
+    retbins: Literal[False] = ...,
+    precision: int = ...,
+    include_lowest: bool = ...,
+    duplicates: Literal["raise", "drop"] = ...,
+    ordered: bool = ...,
+) -> Series: ...
+@overload
+def cut(
+    x: Index | npt.NDArray | Sequence[int] | Sequence[float],
+    bins: int
+    | Series
+    | Int64Index
+    | Float64Index
+    | Sequence[int]
+    | Sequence[float]
+    | IntervalIndex,
+    right: bool = ...,
+    labels: Sequence[Label] | None = ...,
+    retbins: Literal[False] = ...,
+    precision: int = ...,
+    include_lowest: bool = ...,
+    duplicates: Literal["raise", "drop"] = ...,
+    ordered: bool = ...,
+) -> Categorical: ...
+@overload
 def qcut(
-    x, q, labels=..., retbins: bool = ..., precision: int = ..., duplicates: str = ...
-): ...
+    x: Index | npt.NDArray | Sequence[int] | Sequence[float],
+    q: int | Sequence[float] | Series[float] | Float64Index | npt.NDArray,
+    *,
+    labels: Literal[False],
+    retbins: Literal[False] = ...,
+    precision: int = ...,
+    duplicates: Literal["raise", "drop"] = ...,
+) -> npt.NDArray: ...
+@overload
+def qcut(
+    x: Index | npt.NDArray | Sequence[int] | Sequence[float],
+    q: int | Sequence[float] | Series[float] | Float64Index | npt.NDArray,
+    labels: Sequence[Label] | None = ...,
+    retbins: Literal[False] = ...,
+    precision: int = ...,
+    duplicates: Literal["raise", "drop"] = ...,
+) -> Categorical: ...
+@overload
+def qcut(
+    x: Series,
+    q: int | Sequence[float] | Series[float] | Float64Index | npt.NDArray,
+    labels: Literal[False] | Sequence[Label] | None = ...,
+    retbins: Literal[False] = ...,
+    precision: int = ...,
+    duplicates: Literal["raise", "drop"] = ...,
+) -> Series: ...
+@overload
+def qcut(
+    x: Index | npt.NDArray | Sequence[int] | Sequence[float],
+    q: int | Sequence[float] | Series[float] | Float64Index | npt.NDArray,
+    *,
+    labels: Literal[False],
+    retbins: Literal[True],
+    precision: int = ...,
+    duplicates: Literal["raise", "drop"] = ...,
+) -> tuple[npt.NDArray, npt.NDArray[np.float_]]: ...
```

Comment:
first part of tuple is `npt.NDArray[np.intp]`

Reply:
Fixed.